Mathlib/Data/UInt.lean

```diff
@@ -9,6 +9,21 @@ import Mathlib.Algebra.GroupWithZero.Defs
 import Mathlib.Algebra.Ring.Basic
 import Mathlib.Data.ZMod.Defs
 
+/-!
+# Adds Mathlib specific instances to the `UIntX` data types.
+
+The `CommRing` instances (and the `NatCast` and `IntCast` instances from which they is built) are
+scoped in the `UIntX.CommRing` namespace, rather than available globally. As a result, the `ring`
+tactic will not work on `UIntX` types without `open scoped UIntX.Ring`.
+
+This is because the presence of these casting operations contradicts assumptions made by the
+expression tree elaborator, namely that coercions do not form a cycle.
+
+This issue also arises for `CommRing (Fin n)`, but Mathlib has come to rely upon this. The UInt
```

Comment:
```suggestion
The UInt
```

@eric-wieser The `Fin` problem is on my radar.

I don't see any inconsistency with fixing the problem for `UInt*` types and putting off fixing `Fin`. The `Fin` type is arguably mathematical, but the `UInt*` types clearly aren't (they're the integers modulo arbitrary powers of two).

Mentioning `Fin` at all just makes it more likely that we'll have an inconsistency in documentation once the `Fin` problem is fixed. It seems unlikely to me that anyone modifying `Fin` instances would think to look at `Mathlib.Data.UInt`'s module docs.